main/autogen/subclasses.cc

```diff
@@ -92,34 +95,64 @@ optional<Subclasses::SubclassInfo> Subclasses::descendantsOf(const Subclasses::M
+
+    for (auto &n : absl::StrSplit(rawName, "::")) {
+        const auto &nameRef = gs.lookupNameUTF8(n);
+        if (!nameRef.exists())
+            return core::Symbols::noSymbol();
+        sym = gs.lookupClassSymbol(sym, gs.lookupNameConstant(nameRef));
+    }
+    return sym;
+}
+
 // Manually patch the child map to account for inheritance that happens at runtime `self.included`
 // Please do not add to this list.
-void Subclasses::patchChildMap(Subclasses::Map &childMap) {
-    childMap["Opus::SafeMachine"].entries.insert(childMap["Opus::Risk::Model::Mixins::RiskSafeMachine"].entries.begin(),
-                                                 childMap["Opus::Risk::Model::Mixins::RiskSafeMachine"].entries.end());
+void Subclasses::patchChildMap(const core::GlobalState &gs, Subclasses::Map &childMap) {
+    auto riskSafeMachineRef = getConstantRef(gs, "Opus::Risk::Model::Mixins::RiskSafeMachine");
+    childMap[getConstantRef(gs, "Opus::SafeMachine")].entries.insert(childMap[riskSafeMachineRef].entries.begin(),
+                                                                     childMap[riskSafeMachineRef].entries.end());
 }
 
-vector<string> Subclasses::serializeSubclassMap(const Subclasses::Map &descendantsMap,
-                                                const vector<string> &parentNames) {
+vector<string> Subclasses::serializeSubclassMap(const core::GlobalState &gs, const Subclasses::Map &descendantsMap,
+                                                const vector<core::SymbolRef> &parentNames, const bool showPaths) {
     vector<string> descendantsMapSerialized;
+    const auto classFormatString = showPaths ? " class {} {}" : " class {}";
+    const auto moduleFormatString = showPaths ? " module {} {}" : " module {}";
 
-    for (const string &parentName : parentNames) {
-        auto fnd = descendantsMap.find(parentName);
-        if (fnd == descendantsMap.end()) {
+    for (const auto &parentRef : parentNames) {
+        auto fnd = descendantsMap.find(parentRef);
+        if (fnd == descendantsMap.end() || !parentRef.exists()) {
             continue;
         }
         const Subclasses::SubclassInfo &children = fnd->second;
 
+        string parentName =
+            fmt::format("{}", fmt::map_join(symbolName(gs, parentRef),
+                                            "::", [&gs](const core::NameRef &nm) -> string { return nm.show(gs); }));
+
         auto type = children.classKind == ClassKind::Class ? "class" : "module";
         descendantsMapSerialized.emplace_back(fmt::format("{} {}", type, parentName));
 
         auto subclassesStart = descendantsMapSerialized.size();
-        for (const auto &[name, type] : children.entries) {
+        for (const auto &sym : children.entries) {
+            string_view path = sym.loc(gs).file().data(gs).path();
+            // Strip initial "./" from path
+            if (path.find("./") == 0)
+                path = path.substr(2);
```

Comment:
```suggestion
            string_view path = gs.getPrintablePath(sym.loc(gs).file().data(gs).path());
```

`GlobalState::getPrintablePath` respects Sorbet's `--remove-path-prefix` option, which allows stripping arbitrary path prefixes (e.g., it would work if you invoked Sorbet on `--dir=lib`, not just `--dir=.`.

Reply:
We already pass this option in all the relevant places in Stripe's codebase (as far as I can tell). You may wish to double check this anwyays.

Reply:
Is there a way to get the full path? `getPrintablePath` returns the filename.

Reply:
I'm not entirely sure what you mean

https://github.com/sorbet/sorbet/blob/dfd20f3504911c66ce07b0993144b6e7904ce0b9/core/GlobalState.cc#L2221-L2227

The implementation of this method looks basically identical to the code I this `suggestion` replaces?

Reply:
Never mind, the assumption is that the passed in filepath is the root.